File: SamsidParty.Subsonic.Proxy.AppleMusic/Controllers/Ping.cs
using SamsidParty.Subsonic.Common;
using System;
using System.Reflection;
using System.Runtime.InteropServices;

namespace SamsidParty.Subsonic.Proxy.AppleMusic.Controllers
{
    public partial class AppleMusicRequestImplementation : Subsonic.Common.IController
    {
        public async Task<SubsonicResponse> PingAsync()
        {
            return new SubsonicResponse()
            {
                SubsonicResponse1 = GetDefaultResponse()
            };
        }

        public SubsonicSuccessResponse GetDefaultResponse()
        {
            return new SubsonicSuccessResponse()
            {
                Status = SubsonicSuccessResponseStatus.Ok,
                Version = "1.16.1",
                Type = "SamsidParty Subsonic",
                ServerVersion = $"{RuntimeInformation.FrameworkDescription}",
                OpenSubsonic = true,
            };
        }

        public SubsonicFailureResponse GetErrorResponse()
        {
            var defaultRes = GetDefaultResponse();
            return new SubsonicFailureResponse()
            {
                Status = SubsonicFailureResponseStatus.Failed,
                Version = defaultRes.Version,
                Type = defaultRes.Type,
                ServerVersion = defaultRes.Version,
                OpenSubsonic = true,
            };
        }
    }
}

File: FocalSonic/src-vite/src/service/applemusic/ping.ts
import { SubsonicResponse } from "@/types/responses/subsonicResponse";

async function pingInfo() {
    return {
        openSubsonic: true,
        serverVersion: "1.0.0",
        status: "ok",
        type: "applemusic",
        version: "1.16.1",
    } as SubsonicResponse;
}

async function pingView() {
    return true;
}

export const ping = {
    pingInfo,
    pingView,
};
